server/pulp/server/controllers/repository.py

```diff
@@ -186,20 +194,39 @@ def has_all_units_downloaded(repository):
     :return: True if completely downloaded
     :rtype: bool
     """
-    types = {}
-    q_set = model.RepositoryContentUnit.objects(repo_id=repository.repo_id)
-    for association in q_set.only('unit_id', 'unit_type_id'):
-        unit_ids = types.setdefault(association.unit_type_id, [])
-        unit_ids.append(association.unit_id)
-    for type_id, unit_ids in types.items():
-        _model = plugin_api.get_unit_model_by_id(type_id)
-        for page in paginate(unit_ids):
-            q_set = _model.objects(id__in=page, downloaded=False)
-            if q_set.count():
-                return False
+    for qs in _get_mongoengine_repo_querysets(str(repository.repo_id)):
```

Comment:
I think the logic is flipped, so it would need to be `return not any(...)`, which I think highlights how using `any` here is not necessarily better, at least in terms of readability.

I [think](https://github.com/MongoEngine/mongoengine/blob/master/mongoengine/queryset/base.py#L170) that MongoEngine QuerySets will do what you want when evaluated for truthiness here, so assuming that qs is an instance of a MongoEngine QuerySet (which I am), a somewhat more readable option might be `return not any(_get_mongoengine_repo_querysets(str(repository.repo_id))`